cmd/gossamer/configcmd.go

```diff
@@ -300,20 +296,22 @@ func setRPCConfig(ctx *cli.Context, fig *cfg.RPCCfg) {
 
 }
 
-func startRPC(ctx *cli.Context, fig cfg.RPCCfg, apiSrvc *api.Service) *rpc.HTTPServer {
-	if ctx.GlobalBool(RPCEnabledFlag.Name) {
-		return rpc.NewHTTPServer(apiSrvc.API, &json2.Codec{}, fig.Host, fig.Port, fig.Modules)
+func setupRPC(ctx *cli.Context, fig cfg.RPCCfg, stateSrv *state.Service, networkSrvc *network.Service) *rpc.HTTPServer {
+	cfg := &rpc.HTTPServerConfig{
+		BlockApi:   stateSrv.Block,
+		StorageApi: stateSrv.Storage,
+		NetworkApi: networkSrvc,
```

Comment:
I think we should maybe keep network state. See comments in #606.

Reply:
this PR doesn't remove it, it just gets the API needed for rpc directly from network. we could  swap it out later for network service